### PR TITLE
feat: Add API for creating room

### PR DESCRIPTION
**Pull request by nhywieza**
- Implemented API for creating a room
- Added unit tests for creating a room functionality
- Exported `CreateRoomResult` for external use

### CODE REVIEW

File: cozepy/rooms/__init__.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    Room service client.
+    """
+
+    def __init__(self, base_url: str, auth: Auth, requester: Requester):
+        self._base_url = base_url
+        self._auth = auth
+        self._requester = requester
+
+    def create(self, *, bot_id: str, voice_id: str) -> CreateRoomResult:
```
Comment (chyroc) [inline cozepy/rooms/__init__.py:27]: voice_id 可以为空吧

Comment (nhywieza) [reply cozepy/rooms/__init__.py:27]: ok

File: tests/test_room.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import httpx
+import pytest
+
+from cozepy import AsyncCoze, Coze, TokenAuth
+from cozepy.rooms import CreateRoomResult
+
```
Comment (chyroc) [inline tests/test_room.py:6]:
```suggestion
from cozepy.util import random_hex
```

File: tests/test_room.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+@pytest.mark.respx(base_url="https://api.coze.com")
+class TestRooms:
+    def test_sync_rooms_create(self, respx_mock):
+        coze = Coze(auth=TokenAuth(token="token"))
+
```
Comment (chyroc) [inline tests/test_room.py:12]:
```suggestion
room_id = random_hex(10)
room_token = random_hex(10)
app_id = random_hex(10)
uid = random_hex(10)
```

File: tests/test_room.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        respx_mock.post("/v1/audio/rooms").mock(
+            httpx.Response(
+                200,
+                json={
+                    "data": CreateRoomResult(
+                        token="JOIN_ROOM_TOKEN", uid="UID", room_id="ROOM_ID", app_id="APP_ID"
```
Comment (chyroc) [inline tests/test_room.py:18]:
```suggestion
token=room_token, uid=uid, room_id=room_id, app_id=app_id
```

File: tests/test_room.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                    ).model_dump()
+                },
+            )
+        )
+
+        res = coze.rooms.create(bot_id="BOT_ID", voice_id="VOICE_ID")
+        assert res
+        assert res.token == "JOIN_ROOM_TOKEN"
```
Comment (chyroc) [inline tests/test_room.py:26]:
```suggestion
        assert res.token == room_token
```

File: tests/test_room.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        assert res.room_id == "ROOM_ID"
```
Comment (chyroc) [inline tests/test_room.py:27]:
```suggestion
        assert res.room_id == room_id
```

File: tests/test_room.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        assert res.uid == "UID"
```
Comment (chyroc) [inline tests/test_room.py:28]:
```suggestion
        assert res.uid == uid
```

File: tests/test_room.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        assert res.app_id == "APP_ID"
```
Comment (chyroc) [inline tests/test_room.py:29]:
```suggestion
        assert res.app_id == app_id
```

File: tests/test_room.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class TestAsyncRooms:
+    async def test_async_rooms_create(self, respx_mock):
```
Comment (chyroc) [inline tests/test_room.py:33]: 同

File: cozepy/rooms/__init__.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
+from typing import Optional
+
+from cozepy.auth import Auth
+from cozepy.model import CozeModel
+from cozepy.request import Requester
+
+
+class CreateRoomResult(CozeModel):
```
Comment (chyroc) [inline cozepy/rooms/__init__.py:8]: 这个需要在 cozepy/__init__.py export 出去

File: tests/test_room.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
+import httpx
+import pytest
+
+from cozepy import AsyncCoze, Coze, TokenAuth
+from cozepy.rooms import CreateRoomResult
```
Comment (chyroc) [inline tests/test_room.py:5]:
```suggestion
from cozepy import CreateRoomResult
```

**Review by chyroc — APPROVED** [review]

**Pull request merged**